src/NTokenGeometricMean/NTokenGeometricMeanMath.sol

```diff
@@ -114,15 +121,23 @@ function computeNextLiquidity(
     uint256[] memory reserves,
     NTokenGeometricMeanParams memory params
 ) pure returns (uint256 L) {
-    uint256 accumulator;
+    uint256 accumulator = ONE;
     for (uint256 i = 0; i < reserves.length; i++) {
         uint256 a =
             uint256(int256(reserves[i]).powWad(int256(params.weights[i])));
-        if (accumulator != 0) {
-            accumulator.mulWadUp(a);
-        } else {
-            accumulator = a;
-        }
+        accumulator = accumulator.mulWadUp(a);
     }
     return accumulator;
 }
+
+function computeSwapDeltaLiquidity(
+    uint256 amountIn,
+    uint256 reserve,
+    uint256 totalLiquidity,
+    uint256 weight,
+    uint256 swapFee
+) pure returns (uint256) {
+    return weight.mulWadDown(swapFee).mulWadDown(totalLiquidity).mulWadDown(
+        amountIn.divWadDown(reserve)
+    );
```

Comment:
Is there any reason why `G3MMath.computeSwapDeltaLiquidity()` rounds up, but the calculation for `deltaLiquidity` here rounds down?

[G3MMath.sol#L236-L238](https://github.com/primitivefinance/DFMM/blob/fd867a7bd8a28f56e0c7560244e5c195ab12f577/src/GeometricMean/G3MMath.sol#L236-L238)

```solidity
    return weight.mulWadUp(swapFee).mulWadUp(totalLiquidity).mulWadUp(
        amountIn.divWadUp(reserve)
    );
```

I don't think there's a correct direction to round towards, but I'm inclined to think that rounding up is safer as `deltaLiquidity` would round in favor of the protocol and against the user.

Reply:
Yes, obviously, fix is here: https://github.com/primitivefinance/DFMM/pull/119/commits/039093f96d6eb532b414c0c1c380dbeb4690460d.